src/Uno.UI.Composition/Composition/CompositionEffectBrush.skia.cs

```diff
@@ -0,0 +1,1726 @@
+﻿#nullable enable
+
+using System;
+using SkiaSharp;
+using Windows.UI;
+using Microsoft.UI;
+using System.Numerics;
+using Uno.UI.Composition;
+using Windows.Graphics.Effects;
+using Windows.Graphics.Effects.Interop;
+using System.Diagnostics;
+
+namespace Windows.UI.Composition;
+
+public partial class CompositionEffectBrush : CompositionBrush
+{
+	private bool _isCurrentInputBackdrop;
+	private bool? _currentCompMode;
+
+	private SKRect _currentBounds;
+	private SKImageFilter? _filter;
+
+	internal bool HasBackdropBrushInput { get; private set; }
+
+	internal bool UseBlurPadding { get; set; }
+
+	private SKImageFilter? GenerateEffectFilter(object? effect, SKRect bounds, bool tryUsingSourceSize = false)
+	{
+		// TODO: https://user-images.githubusercontent.com/34550324/264485558-d7ee5062-b0e0-4f6e-a8c7-0620ec561d3d.png
+		// TODO: Cache pixel shaders (see dwmcore.dll!CCompiledEffectCache), needed in order to implement animations and online rendering
+
+		switch (effect)
```

Comment:
Let's refactor all switch case statements in separate methods for readability, this method is too large.